django/db/migrations/serializer.py

```diff
@@ -246,11 +247,13 @@ class RegexSerializer(BaseSerializer):
     def serialize(self):
         imports = {"import re"}
         regex_pattern, pattern_imports = serializer_factory(self.value.pattern).serialize()
-        regex_flags, flag_imports = serializer_factory(self.value.flags).serialize()
+        # re.UNICODE is implicit flag, hence we need to turn it off
+        flags = self.value.flags ^ re.UNICODE
```

Comment:
All compiled regex will be serialized with `re.UNICODE` flag e.g.:
```
re.compile(r'^\w+$') -> re.compile(r'^\w+$', 32)
```
instead of `re.compile(r'^\w+$')`.

That breaks tests with `assertSerializedEqual` (`migrations.test_writer.WriterTests`) e.g.
`test_serialize_compiled_regex`, `test_serialize_class_based_validators`, `test_serialize_lazy_objects`